server/src/main/java/org/elasticsearch/cluster/desirednodes/DesiredNodesSettingsValidator.java

```diff
@@ -23,19 +23,26 @@
 public class DesiredNodesSettingsValidator {
     private record DesiredNodeValidationError(int position, @Nullable String externalId, RuntimeException exception) {}
 
+    @Deprecated(forRemoval = true)
```

Comment:
What is this going to be replaced with?

Reply:
Hopefully, strict equality (`String.equals`)

Reply:
Or, based on the answers to [this question above](https://github.com/elastic/elasticsearch/pull/100133#issuecomment-1743392719), with nothing at all (no checks)